src/hooks/network/useTryConnect.tsx

```diff
@@ -25,15 +25,25 @@ const attemptToConnectToEngine = async ({
   videoRef,
   setIsSceneReady,
   settingsEngine,
+  timeToConnect,
 }: {
   authToken: string
   videoWrapperRef: React.RefObject<HTMLDivElement>
   setAppState: (newAppState: Partial<ReturnType<typeof useAppState>>) => void
   videoRef: React.RefObject<HTMLVideoElement>
   setIsSceneReady: React.Dispatch<React.SetStateAction<boolean>>
   settingsEngine: SettingsViaQueryString
+  timeToConnect: number
 }) => {
   const connection = new Promise<boolean>((resolve, reject) => {
+    const cancelTimeout = setTimeout(() => {
+      EngineDebugger.addLog({
+        label: 'useTryConnect.tsx',
+        message: 'Took too long to connect, cancelTimeout',
+      })
+      reject('took too long to connect, cancelTimeout')
```

Comment:
Does the user see this as a toast? If so I would suggest:
```suggestion
      reject(`Took too long to connect, timed out.`)
```

Reply:
No the user will not see this as a toast.